packages/styletron-engine-snapshot/src/create-render.js

```diff
@@ -0,0 +1,46 @@
+// @flow
+
+import {
+  type StyleObject,
+  type FontFaceObject,
+  type KeyframesObject,
+} from "styletron-standard";
+
+import stringify from "json-stable-stringify";
+
+/**
+  @see: https://prettier.io/docs/en/browser.html
+ */
+import prettier from "prettier/standalone";
+import babylonParser from "prettier/parser-babylon";
+
+/*
+This adds a "deterministic" engine that simply returns a `JSON.stringify`-ed
+version of the style object (with aplha-sorted keys).
+*/
+export default function createRender(kind: string) {
+  return function render(
+    obj: ?(StyleObject | FontFaceObject | KeyframesObject),
+  ): string {
+    if (!obj) {
+      return "";
+    }
+
+    return prettier
+      .format(`${kind} = ${stringify(obj)}`, {
+        // babylon is treated as 'babel'
+        parser: "babel",
+        plugins: [babylonParser],
+
+        // don't add a semicolon at the end of the declaration
+        semi: false,
+
+        // default/prefer use of `'` for strings when needed
+        singleQuote: true,
+
+        // force newline insertions more often
+        printWidth: 40,
+      })
+      .replace(`${kind} = `, `${kind}=`);
```

Comment:
Rather than this replacement, it might be preferable to use the [prettier json5 parser](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEwA6yCGAzeAnDJAAnXGnlkKIzVpAAk4AbJiIgdQjyYBNaaQAXwA01MqwLIxeOHxCiMALwCSUHnAAeVDAFoAjAAYjGQfJAQADjACW0AM7JQmPHggB3AArOEDlJgBuENY8ZgBGeJhgANZwMADKFpHWUADmyNiYTHZwwiAAFjAAtkzsedbwdolgcHE+5db+5QCeyOB2DrnJ2XgwHhEphZjpmdm5AFZ2GgBCEdGxcZiFcAAyyXDDWTkgExpxySlMcACKAK4Q8BujIIl43a0T0ACsZhZ4yTDswTB5yAAsBrlXhBsuwIhZWpk3Jgmh0QABHM7wPqWXwgTB2HRQOCyWRmGQI6wyPqYAZDJAZTa5bKFazIGB4E5bOz7Q6nc7rJD0xm5GCYUKfHjfZAAJh5EWsTH2AGEIIVBq0oNB1rkTtkACp83wU7KCQRAA) and then concatenate whatever prefix you want

Reply:
ooh nice.

Using the `json5` parser + string prefixing is definitely more legible and less `hacky`. It's probably also more performant/faster since it's a much simpler grammar than full fledged js.

Unfortunately it turns out that there is no standalone `json5` parser and it is instead [bundled s part of `babylon`](https://github.com/prettier/prettier/blob/a093bb3f7b9f59d8cbaf7e20f97f6fafceaef21b/src/common/internal-plugins.js#L28), so we wont see the build get smaller 🙁

Reply:
Ah, oh well 🤷‍♂